src/tests/integration/utils.py

```diff
@@ -328,12 +398,35 @@ def check_endpoint(self, endpoint, resource, response_code,
 
         nullable_fields = [] if nullable_fields is None else nullable_fields
         schema = self.get_resource_schema(resource)
-        response = self.make_request(endpoint, response_code,
-                                     params=query_params)
+        response = self.make_request(endpoint,
+                                     response_code,
+                                     params=query_params,
+                                     max_elapsed_seconds=12)
```

Comment:
Is there any request take ~10s?

Reply:
There is a lot of variance especially when running this many tests. Usually all tests pass with 5 seconds but it would make jenkins yell at us too often.

Reply:
I think 10 secs are too slow IMO, I'd recommend using 5 sec for now and let's see how often the integration test will break then adjust base on the result. Do you know which component takes the longest time to response?

Reply:
There wasn't any specific endpoint that would cause failures. Seems like one query would just hang for around 10 seconds usually. Sometimes 20+ seconds. I just ran some tests with it at 5 seconds and it worked fine so I'll set it back to 5